genesyscloud/user/resource_genesyscloud_user.go

```diff
@@ -0,0 +1,287 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/constants"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+)
+
+type AgentUtilizationWithLabels struct {
+	Utilization       map[string]util.MediaUtilization `json:"utilization"`
+	LabelUtilizations map[string]util.LabelUtilization `json:"labelUtilizations"`
+	Level             string                           `json:"level"`
+}
+
+func GetAllUsers(ctx context.Context, sdkConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
```

Comment:
Does this need to be public?

Reply:
yes, it is used inside resource_genesyscloud_user_roles_schema.go